src/components/Navigation/SearchBar/SearchBar.vue

```diff
@@ -0,0 +1,264 @@
+    DownArrowIcon,
+    SearchIcon,
+    SwitchIcon
+  },
+  setup(props: any) {
+    const store = useStore();
+
+    const isLiActive = ref<any>(false);
+    const filterVal = ref("");
+    const orderVal = ref<any>(computed(() => store.state.nav.foodListOrder));
+    const orderOptions: OrderOption[] = [
+      {
+        category: "Price",
+        order: "Low to High",
+        type: "price_asc"
+      },
+      {
+        category: "Price",
+        order: "High to Low",
+        type: "price_desc"
+      },
+      {
+        category: "Alphabetically",
+        order: "A to Z",
+        type: "alph_asc"
+      },
+      {
+        category: "Alphabetically",
+        order: "Z to A",
+        type: "alph_desc"
+      }
+    ];
+
+    const changeFoodListView = function(): void {
+      store.commit("nav/toggleFoodListView")
+    };
+    const toggleOrderList = function(): void {
+      isLiActive.value = !isLiActive.value;
+    };
+
+    const setFoodOrder = function(event: any): void{
+      const id = event.target.closest(".np-search-bar__filter-list-item").id;
+      const activeOrder = orderOptions.filter((option: OrderOption) => option.type === id);
+      store.commit("nav/setFoodListOrder", activeOrder[0]);
+      store.commit("nav/orderFoodList");
+      toggleOrderList();
+    };
+
+    const setActiveFilter = function(): void {
+      store.commit("nav/filterFoodList", filterVal.value);
+      store.commit("nav/")
+    };
+
+    return {
+      changeFoodListView,
+      setActiveFilter,
+      setFoodOrder,
+      toggleOrderList,
+      filterVal,
+      orderVal,
+      orderOptions,
+      isLiActive
+    };
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+.np-search-bar {
+  @apply flex flex-row justify-start items-center w-full mb-10;
+  height: 56px;
+  background-color: #ffffff;
+
+  svg {
+    height: inherit;
+    width: auto;
+    fill: theme("colors.secondary.DEFAULT");
+  }
+
+  &__toggle-list-view {
+    height: inherit;
+    outline: none;
+
+    svg {
+      padding: 10px 20px 10px 0;
+
+      &:hover {
+        fill: theme("colors.primary.DEFAULT");
+      }
+    }
+  }
+
+  &__input,&__filter{
+    height: inherit;
+    margin-left: 44px;
+    border-radius: 2px;
+    border: solid 1px theme("colors.secondary.lighter");
+  }
+
+  &__input {
+    @apply flex flex-row justify-start items-center w-full;
+
+    height: inherit;
+    padding: 10px;
+
+    &:hover {
+      border-color: theme("colors.primary.DEFAULT");
+    }
+
+    &-el {
+      width: 100%;
+      font-size: 19px;
+      outline: none;
+      color: theme("colors.secondary.dark");
+    }
+    svg {
+      height: 36px;
+      margin-right: 20px;
+    }
+  }
+
+  &__filter {
+    width: 300px;
+    height: inherit;
+
+    &:hover {
+      border-color: theme("colors.primary.DEFAULT");
+    }
+
+    &-option {
+      @apply flex flex-row justify-between items-center;
+
+      height: inherit;
+      width: inherit;
+      padding: 16px;
+
```

Comment:
trzymaj bloki które są jednością razem. Popraw to w paru miejscach bo masz takie przerwy czasami